tests/libtest/lib1960.c

```diff
@@ -0,0 +1,143 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.haxx.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ * SPDX-License-Identifier: curl
+ *
+ ***************************************************************************/
+#include "test.h"
+
+#ifdef WIN32
+#undef _WIN32_WINNT
```

Comment:
Hmm, seems like my comment got applied only to the commit and not shown in the PR:
Or maybe just make HAVE_INET_PTON make a requirement for executing this test? The Windows headers might already be included indirectly by test.h, so this might be too late to redefine the value.

Reply:
ah hurt by that *again* ...

Reply:
but then, where/when is mingw.h included which is the header that apparently sets the "bad" define?

Reply:
According to: https://github.com/msys2/MINGW-packages/issues/6191

Reply:
While here: should we perhaps drop mingw v1 support and live a happier life?

Reply:
trying a precheck to skip the test if inet_pton is not present

Reply:
Replying to https://curl.se/mail/lib-2023-03/0006.html:

"original" MinGW has been a constant source of issues :( (I've spent many days digging for issue unique to it) Its Windows headers are seriously outdated. The compiler as well. It is (and always was) difficult to install. It only support Intel 32-bit, which by itself is an almost dead platform. Now even its original website (mingw.org) is gone.

It's never an easy decision to drop a platform, compiler, feature, but in this case I'd vote to do so. It takes off a ton of complications, leaving largely a modern mingw-w64 and MSVC to worry about on Windows.

[ Meanwhile mingw-w64 can be installed via MSYS2 or standalone, or using most unixy package managers as well. Its Wine headers are up to date. It's actively developed. ]